spec/type.dd

```diff
@@ -548,36 +548,62 @@ values `false` and `true`, respectively. Casting an expression to `bool` means
 testing for `0` or `!=0` for arithmetic types, and `null` or `!=null` for
 pointers or references.)
 
-$(H2 $(LNAME2 delegates, Delegates))
 
-$(P Delegates are an aggregate of two pieces of data: an
-object reference and a pointer to a non-static member function, or a pointer to
-a closure and a pointer to a nested function. The object reference forms the
-`this` pointer when the function is called.)
+$(H2 $(LNAME2 functions, Function Types))
 
-$(P Delegates are declared similarly to function pointers:)
+$(P A function type has the form:)
+
+$(GRAMMAR
+$(GLINK TypeCtor)$(OPT) $(GLINK BasicType) $(GLINK2 function, Parameters) $(GLINK2 function, FunctionAttributes)$(OPT)
+)
+
+$(P A function type e.g. `int(int)` is only used for type tests.
+A function type $(DDSUBLINK spec/declaration, alias-function, can be aliased).)
```

Comment:
I have a serious reserve on this change. It seems to bring the special syntax allowed only for AliasDecl. But in other context that syntax can create ambiguities. For example

```
S(T,U)
```

requires work during expsema, as it can be a ctor call or a function type

Reply:
well, to go further, it's a bit like `T[2]` which can be either an array Type, an IndexExp, or an attempt to statically evaluate the element of a template parameter.

So finally, you should think more about that. Does this really work ?

Reply:
Function type syntax is only allowed in a type context, so that rules out a call expression.